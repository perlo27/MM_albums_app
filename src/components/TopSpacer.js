import React from 'react';
import { View, StyleSheet, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';

export const TopSpacer = () =>
  Platform.OS === 'android' ? null : <View style={styles.topSpacer} />;

const styles = StyleSheet.create({
  topSpacer: {
    height: getStatusBarHeight(),
  },
});
