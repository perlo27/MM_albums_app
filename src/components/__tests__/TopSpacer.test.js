import React from 'react';
import renderer from 'react-test-renderer';
import { TopSpacer } from '../TopSpacer';



describe('TopSpacer', () => {
  it('should render correctly on ios', () => {
    const tree = renderer
      .create(<TopSpacer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly on android', () => {
    jest.mock('Platform', () => ({
      OS: 'android',
    }));
    const tree = renderer
      .create(<TopSpacer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});