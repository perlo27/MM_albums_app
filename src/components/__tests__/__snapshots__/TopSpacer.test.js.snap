// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSpacer should render correctly on android 1`] = `null`;

exports[`TopSpacer should render correctly on ios 1`] = `
<View
  style={
    Object {
      "height": 20,
    }
  }
/>
`;
